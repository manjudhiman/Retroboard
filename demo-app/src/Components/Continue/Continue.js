import React from 'react';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';
import './Styles/Continue.css';

const Continue = (props) => {

  return (
    <div className="text-box">
    <h3>To Continue </h3>
      <Button variant="outline-dark" onClick={props.addHandler}>   +   </Button>
    {props.texts(props.notes, 'continue')}
    </div>
  );
}

Continue.propTypes = {
  texts: PropTypes.func,
  notes: PropTypes.object
}

export default Continue;
