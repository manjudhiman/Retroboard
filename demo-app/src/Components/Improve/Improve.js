import React from 'react';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';
import './Styles/Improve.css';

const Improve = (props) => {

  return (
    <div className="text-box">
    <h3>To Improve </h3>
    <Button variant="outline-dark" onClick={props.addHandler}>   +   </Button>
       {props.texts(props.notes, 'improve')}
    </div>
  );
}

Improve.propTypes = {
  texts: PropTypes.func,
  notes: PropTypes.object
}

export default Improve;
