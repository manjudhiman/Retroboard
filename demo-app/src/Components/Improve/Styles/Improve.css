.text-box {
  width: 250px;
  margin: 15px;
}
