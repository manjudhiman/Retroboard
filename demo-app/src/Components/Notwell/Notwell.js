import React from 'react';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';
import './Styles/Notwell.css';

const Notwell = (props) => {

  return (
    <div className="text-box">
    <h3>Didn't Go Well </h3>
    <Button variant="outline-dark" onClick={props.addHandler}>   +   </Button>
    {props.texts(props.notes, 'notwell')}
    </div>
  );
}

Notwell.propTypes = {
  texts: PropTypes.func,
  notes: PropTypes.object
}

export default Notwell;
