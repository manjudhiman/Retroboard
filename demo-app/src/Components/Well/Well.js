import React from 'react';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';
import './Well.css';

const Well = (props) => {
  return (
    <div className="text-box">
    <h3>Went Well </h3>
    <Button variant="outline-dark" onClick={props.addHandler}>   +   </Button>
      {props.texts(props.notes, 'well')}
    </div>
  );
}

Well.propTypes = {
  texts: PropTypes.func,
  notes: PropTypes.object
}

export default Well;
