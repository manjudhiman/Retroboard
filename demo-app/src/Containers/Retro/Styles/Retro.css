.test1 {
  width: 25%;
}

.container {
  display: inline-flex;
}

.save {
  height: 50px;
  width: 100px;
  display: inline-flex;
}
 .banner {
   color: "#FFF";
   background-color: "green"
 }
