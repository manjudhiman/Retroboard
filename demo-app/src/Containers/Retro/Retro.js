import React, { useState, useEffect, fragment} from 'react';
import Notwell from '../../Components/Notwell/Notwell';
import Well from '../../Components/Well/Well';
import Improve from '../../Components/Improve/Improve';
import Continue from '../../Components/Continue/Continue';
import Sticky from '../../Components/Sticky/Sticky';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import Banner from 'react-js-banner';
import './Retro.css'

const Retro = () => {
  const [stickyPoints, updateStickyPoints] = useState(
      {  well: null,
        notwell: null,
        continue: null,
        improve: null
  }
  );

  const [error, updateError] = useState(false)
  const [dataLoaded, updateDataLoadedState] = useState(false)
  useEffect(() => {
    // This is required to bypass the cors-origin request
    var proxyUrl = 'https://aqueous-fjord-87609.herokuapp.com/',
    targetUrl = 'https://fast-brook-22761.herokuapp.com/retro_info?retro=2'
    axios.get(proxyUrl + targetUrl)
    .then(response => response.data)
    .then(data => {
      updateStickyPoints(data)
  });
  },[]);

  const addHandler = function(key){
    const points = {...stickyPoints}
    const p = points[key]
    let keys = Object.keys(p)
    let last_key = parseInt(keys[keys.length-1])
    points[key][(last_key+1).toString(10)] = {
      value: '',
    }
    updateStickyPoints(
      points
    )
  }

  const updateText = (index,text, k) => {
    let points = {...stickyPoints}

    points[k][index].value = text;
    updateStickyPoints(
      points
    )
  }

  const texts = (points, k) => {
    if (points == null) {
      return
    }

    return Object.keys(points).map((key, id) => {
      console.log("key ", key)
      console.log("id  ", id)
      return <Sticky text={points[key]} key={id} changed={(e) => updateText(key,e.target.value, k) }/>
    })
  };

  const saveButton =() => {
    const json2 = JSON.stringify({...stickyPoints})
    json2.replace(/".+?"/g, s => s.toString())
    // This is required to bypass the cors-origin error.
    var proxyUrl = 'https://aqueous-fjord-87609.herokuapp.com/',
    targetUrl = 'https://fast-brook-22761.herokuapp.com/retro_info?retro=2'
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },

    };
    axios.post(proxyUrl + targetUrl, json2, requestOptions)
    .then(response => {
      console.log("response ", response.status);
      if (response.status === '204') {
        console.log("Saved!")
        updateDataLoadedState(true)

        updateError(false)
      }
    })
    .catch(function(error){
      updateError(true)
    })
  };

 let banner = <div> </div>
 if (dataLoaded) {
    banner =  <Banner title="Changes Saved!!" className="banner" visibleTime={1000}/>
  }

 let savebtn = <Button className="save" variant="primary" size="sm" onClick={saveButton}>Try Again!</Button>
 if (!error) {
    savebtn  = <Button className="save" variant="primary" size="sm" onClick={saveButton }>Save</Button>
  }

  return(
    <React.Fragment>
      {banner}
      <div className="container">
        <Well props={stickyPoints.well} addHandler={() => addHandler('well')} texts={()=>texts(stickyPoints.well, 'well')}/>
        <Notwell props={stickyPoints.notwell} addHandler={() => addHandler('notwell')} texts={()=>texts(stickyPoints.notwell,'notwell')}/>
       <Improve props= {stickyPoints.improve} addHandler={() => addHandler('improve')} texts={()=>texts(stickyPoints.improve, 'improve')}/>
        <Continue props={stickyPoints.continue} addHandler={() => addHandler('continue')} texts={()=>texts(stickyPoints.continue, 'continue')}/>
      </div>
      <div>
        {savebtn}
      </div>
    </React.Fragment>
  )
}

export default Retro;
