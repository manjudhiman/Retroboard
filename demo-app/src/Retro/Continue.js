import React, { useState } from 'react';
import { IoIosAdd } from "react-icons/io";
import Sticky from './Sticky';
import './Well.css';

const Continue = () => {
  const [stickyPoints, updateStickyPoints] = useState(
    {
      0: {
        value: '',
      }
    }
  );

  const addHandler = function(){
    const points = {...stickyPoints}
    let keys = Object.keys(points)
    console.log("**keys", keys)
    let last_key = parseInt(keys[keys.length-1])
    points[last_key+1] = {
      value: '',
    }
    updateStickyPoints(
      points
    )
  }

  const updateText = (index,text) => {
    let points = {...stickyPoints}
    points[index].value = text;
    updateStickyPoints(
      points
    )
  }

  const texts = (points) => {
    console.log("*points**", points)
    return Object.keys(points).map((key) => {
      return <Sticky text={points[key]} changed={(e) => updateText(key,e.target.value) }/>
    })
  };

  return (
    <div className="well">
    <h3 className="text-css">To Continue </h3>
    <IoIosAdd onClick={addHandler} />
    {texts(stickyPoints)}
    </div>
  );
}



export default Continue;
