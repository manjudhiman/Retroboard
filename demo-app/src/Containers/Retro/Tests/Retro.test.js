import React from 'react';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Retro from '../Retro';
import Notwell from '../../../Components/Notwell/Notwell';
import Well from '../../../Components/Well/Well';
import Improve from '../../../Components/Improve/Improve';
import Continue from '../../../Components/Continue/Continue';

configure({adapter: new Adapter()});

describe('<Retro />' , () => {
  it('renders correctly', () => {
  const tree = renderer
    .create(<Retro />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


describe('renders ', () => {
  const wrapper = mount(<Retro />)
  it('Well component', () => {
    expect(wrapper.find(Well)).toBeTruthy();
  })
  it('Didnt well component', () => {
    expect(wrapper.find(Notwell)).toBeTruthy();
  })
  it('Improve component', () => {
    expect(wrapper.find(Improve)).toBeTruthy();
  })
  it('Continue component', () => {
    expect(wrapper.find(Continue)).toBeTruthy();
  })
})

});
