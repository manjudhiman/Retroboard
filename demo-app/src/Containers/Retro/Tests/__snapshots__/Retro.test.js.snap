// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Retro /> renders correctly 1`] = `
Array [
  <div>
     
  </div>,
  <div
    className="container"
  >
    <div
      className="text-box"
    >
      <h3>
        Went Well 
      </h3>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
           +   
      </button>
    </div>
    <div
      className="text-box"
    >
      <h3>
        Didn't Go Well 
      </h3>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
           +   
      </button>
    </div>
    <div
      className="text-box"
    >
      <h3>
        To Improve 
      </h3>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
           +   
      </button>
    </div>
    <div
      className="text-box"
    >
      <h3>
        To Continue 
      </h3>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
           +   
      </button>
    </div>
  </div>,
  <div>
    <button
      className="save btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>,
]
`;
